skimage/measure/profile.py

```diff
@@ -109,3 +122,222 @@ def _line_profile_coordinates(src, dst, linewidth=1):
     perp_cols = np.array([np.linspace(col_i - col_width, col_i + col_width,
                                       linewidth) for col_i in line_col])
     return np.array([perp_rows, perp_cols])
+
+
+def _line_profile_coordinates3d(src, dst, linewidth=1):
+    """Return the coordinates of the profile of an image along a scan line.
+
+    Parameters
+    ----------
+    src : 3-tuple of numeric scalar (float or int)
+        The start point of the scan line.
+    dst : 3-tuple of numeric scalar (float or int)
+        The end point of the scan line.
+    linewidth : int, optional
+        Width of the scan, perpendicular to the line,
+        since the line is 3d, this value is the diameter
+        of a 3d cylinder following the scan line.
+
+    Returns
+    -------
+    coords : array, shape (3, N, C), float
+        The coordinates of the profile along the scan line. The length of the
+        profile is the ceil of the computed length of the scan line.
+
+    Notes
+    -----
+    This is a utility method meant to be used internally by skimage functions.
+    The destination point is included in the profile, in contrast to
+    standard numpy indexing.
+    """
+    src_row, src_col, src_dim3 = src = np.asarray(src, dtype=float)
+    dst_row, dst_col, dst_dim3 = dst = np.asarray(dst, dtype=float)
+    d_row, d_col, d_plane = dst - src
+
+    # Get one unit vector perpendicular to direction vector to find a point
```

Comment:
Can you try to explain this with a figure somewhere? I can't figure out what exactly you're doing here...